maildir/maildir.c

```diff
@@ -1609,6 +1611,57 @@ static enum MailboxType maildir_path_probe(const char *path, const struct stat *
   return MUTT_UNKNOWN;
 }
 
+/**
+ * maildir_check_dir - create a new maildir directory tree
+ * @param path Path to create the maildir under
+ * @retval -1 Creation failed, we attempt to cleanup anything we already created
+ * @retval  0 Creation success
+ *
+ */
+static int maildir_create_maildir(const char *path)
+{
+  char tmp[PATH_MAX];
+
+  for (int i = 0; i < mutt_array_size(maildir_subdirs); i++)
+  {
+    snprintf(tmp, sizeof(tmp), "%s/%s", path, maildir_subdirs[i]);
+
+    if (mutt_file_mkdir(tmp, S_IRWXU) != 0)
```

Comment:
Per our conversation on IRC: we probably want to prompt users in the case of `EEXIST` here. Maybe this is best done in `maildir_create_mailbox` with e.g. `maildir_path_probe`? To keep this function purely for creation

Reply:
Yes.  Ideally the 'business' decisions should be made by the front-end code.

From the Browser, we might see:
- User starts browser: `mutt_buffer_select_file()`
- User calls function `<create-mailbox>`
  - Prompt: "Enter name:"
- Browser calls `mx_mbox_create()`
- The MXAPI
  - selects backend based on the `Account`
  - calls `maildir_mbox_create()`
- The backend returns `EEXISTS`
- `mx_mbox_create()` relays the error back to the browser
- Browser tells the user "Folder 'X' already exists"